test/distributed/org/apache/cassandra/distributed/mock/nodetool/InternalNodeProbe.java

```diff
@@ -0,0 +1,202 @@
+import org.apache.cassandra.db.HintedHandOffManager;
+import org.apache.cassandra.db.Keyspace;
+import org.apache.cassandra.db.compaction.CompactionManager;
+import org.apache.cassandra.gms.FailureDetector;
+import org.apache.cassandra.gms.FailureDetectorMBean;
+import org.apache.cassandra.gms.Gossiper;
+import org.apache.cassandra.locator.EndpointSnitchInfo;
+import org.apache.cassandra.locator.EndpointSnitchInfoMBean;
+import org.apache.cassandra.metrics.CassandraMetricsRegistry;
+import org.apache.cassandra.metrics.DefaultNameFactory;
+import org.apache.cassandra.metrics.MetricNameFactory;
+import org.apache.cassandra.net.MessagingService;
+import org.apache.cassandra.service.CacheService;
+import org.apache.cassandra.service.CacheServiceMBean;
+import org.apache.cassandra.service.GCInspector;
+import org.apache.cassandra.service.StorageProxy;
+import org.apache.cassandra.service.StorageService;
+import org.apache.cassandra.streaming.StreamManager;
+import org.apache.cassandra.tools.NodeProbe;
+import org.apache.cassandra.tools.RepairRunner;
+
+public class InternalNodeProbe extends NodeProbe
+{
+    // dummy constructors
+    public InternalNodeProbe(String host, int port, String username, String password) throws IOException
+    {
+        super(host, port, username, password);
+    }
+
+    public InternalNodeProbe(String host, int port) throws IOException
+    {
+        super(host, port);
+    }
+
+    protected void connect() throws IOException
+    {
+        // note that we are not connecting via JMX for testing
+        mbeanServerConn = null;
+        jmxc = null;
+
+        ssProxy = StorageService.instance;
+        msProxy = MessagingService.instance();
+        streamProxy = StreamManager.instance;
+        compactionProxy = CompactionManager.instance;
+        fdProxy = (FailureDetectorMBean) FailureDetector.instance;
+        cacheService = CacheService.instance;
+        spProxy = StorageProxy.instance;
+        hhProxy = HintedHandOffManager.instance;
+        gcProxy = new GCInspector();
+        gossProxy = Gossiper.instance;
+        memProxy = ManagementFactory.getMemoryMXBean();
+        runtimeProxy = ManagementFactory.getRuntimeMXBean();
+    }
+
+    @Override
+    public void close() throws IOException
+    {
+        // nothing to close. no-op
+    }
+
+    @Override
+    // overrides all the methods referenced mbeanServerConn/jmxc in super
+    public EndpointSnitchInfoMBean getEndpointSnitchInfoProxy()
+    {
+        return new EndpointSnitchInfo();
+    }
+
+    @Override
+    public CacheServiceMBean getCacheServiceMBean()
+    {
+        return cacheService;
+    }
+
+    @Override
+    public void repairAsync(final PrintStream out, final String keyspace, Map<String, String> options) throws IOException
+    {
+        RepairRunner runner = new RepairRunner(out, ssProxy, keyspace, options);
+        try
+        {
+            ssProxy.addNotificationListener(runner, null, null);
```

Comment:
can we refactor `NodeProbe` so `ssProxy` is a function call?  That way this method doesn't need to be overridden